src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/ResponseBuilderV2.kt

```diff
@@ -0,0 +1,159 @@
+package it.pagopa.ecommerce.helpdesk.utils
+
+import it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationCompletedData
+import it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationRequestData
+import it.pagopa.ecommerce.commons.documents.v2.TransactionUserReceiptData
+import it.pagopa.ecommerce.commons.documents.v2.authorization.NpgTransactionGatewayAuthorizationData
+import it.pagopa.ecommerce.commons.documents.v2.authorization.PgsTransactionGatewayAuthorizationData
+import it.pagopa.ecommerce.commons.domain.v2.TransactionWithClosureError
+import it.pagopa.ecommerce.commons.domain.v2.pojos.*
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OperationResultDto
+import it.pagopa.ecommerce.commons.generated.server.model.AuthorizationResultDto
+import it.pagopa.ecommerce.commons.utils.v2.TransactionUtils.getTransactionFee
+import it.pagopa.generated.ecommerce.helpdesk.model.*
+import it.pagopa.generated.ecommerce.nodo.v2.model.UserDto
+import java.util.*
+
+fun baseTransactionToTransactionInfoDtoV2(baseTransaction: BaseTransaction): TransactionResultDto {
+    val amount = baseTransaction.paymentNotices.sumOf { it.transactionAmount.value }
+    val fee = getTransactionFees(baseTransaction).orElse(0)
+    val totalAmount = amount.plus(fee)
+    val transactionAuthorizationRequestData = getTransactionAuthRequestedData(baseTransaction)
+    val transactionAuthorizationCompletedData = getTransactionAuthCompletedData(baseTransaction)
+    val transactionUserReceiptData = getTransactionUserReceiptData(baseTransaction)
+
+    // Build user info
+
+    val userInfo =
+        UserInfoDto()
+            // TODO to be valued here with PDV integration
+            .notificationEmail("")
+            // TODO this field is statically valued with GUEST eCommerce side into Nodo ClosePayment
+            // requests. Must be populated dinamically when logic will be updated eCommerce side
+            // (event-dispatcher/transactions-service)
+            .authenticationType(UserDto.TypeEnum.GUEST.toString())
+    // build transaction info
+    val transactionInfo =
+        TransactionInfoDto()
+            .creationDate(baseTransaction.creationDate.toOffsetDateTime())
+            .status(getTransactionDetailsStatus(baseTransaction))
+            .statusDetails(transactionAuthorizationCompletedData?.authorizationCode)
```

Comment:
In main version we value this field with [PGS errorCode](https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/blob/53b499a0b3c16295b87c684972164433f848a498/src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/ResponseBuilder.kt#L109C13-L109C77) (if any).
This can be refactored as

```suggestion
            .statusDetails(getStatusDetail(transactionAuthorizationCompletedData?.transactionGatewayAuthorizationData))
```

Using the following function:
```
fun getStatusDetail(transactionGatewayAuthorizationData: TransactionGatewayAuthorizationData?): String? =
    when (transactionGatewayAuthorizationData) {
        is PgsTransactionGatewayAuthorizationData -> transactionGatewayAuthorizationData.errorCode
        is NpgTransactionGatewayAuthorizationData -> null
        else -> null
    }
```
Actually we don't have a mapping for PGS -> NPG error code field so the above snippet set it to null explicitly in order to have a point to handle this field